packages/web-runtime/src/container/api.ts

```diff
@@ -0,0 +1,266 @@
+import VueRouter, { RouteConfig } from 'vue-router'
+import clone from 'lodash-es/clone'
+import {
+  ClassicApplicationScript,
+  RuntimeApi,
+  ApplicationNavigationItem,
+  ApplicationQuickActions
+} from './types'
+import { ApiError } from './error'
+import { get, isEqual, isObject, isArray } from 'lodash-es'
+import { Store } from 'vuex'
+import Vue, { Component } from 'vue'
+import { Wormhole } from 'portal-vue'
+
+/**
+ * inject application specific routes into runtime
+ *
+ * @param applicationId
+ * @param router
+ * @param routes
+ */
+const announceRoutes = (applicationId: string, router: VueRouter, routes: RouteConfig[]): void => {
+  if (!isArray(routes)) {
+    throw new ApiError("routes can't be blank")
+  }
+
+  const applicationRoutes = routes.map(applicationRoute => {
+    if (!isObject(applicationRoute)) {
+      throw new ApiError("route can't be blank", applicationRoute)
+    }
+
+    const route = clone(applicationRoute)
+    route.name = applicationId === route.name ? route.name : `${applicationId}-${route.name}`
+    route.path = `/${encodeURI(applicationId)}${route.path}`
+
+    if (route.children) {
+      route.children = route.children.map(childRoute => {
+        if (!isObject(applicationRoute)) {
+          throw new ApiError("route children can't be blank", applicationRoute, childRoute)
+        }
+
+        const route = clone(childRoute)
+        route.name = `${applicationId}-${childRoute.name}`
+        return route
+      })
+    }
+
+    return route
+  })
+
+  router.addRoutes(applicationRoutes)
+}
+
+/**
+ * inject application specific navigation items into runtime
+ *
+ * @param applicationId
+ * @param store
+ * @param navigationItems
+ */
+const announceNavigationItems = (
+  applicationId: string,
+  store: Store<unknown>,
+  navigationItems: ClassicApplicationScript['navItems']
```

Comment:
Why not 
```suggestion
  navigationItems: ApplicationNavigationItem[]
```
? 🤔

Reply:
Or in other words: you already introduced types, why not use them here? Instead you're creating a dependency to the `ClassicApplicationScript` which should not be needed in the (more) generic runtime API.